views/js/pciCreator/dev/audioRecordingInteraction/runtime/audioRecordingInteraction.js

```diff
@@ -0,0 +1,720 @@
+             * The asynchronous nature of this operation might also be problematic for saving the recording
+             * when the user exit the item in the middle of a recording (destroy() on the PCI interface should be implemented as a Promise)
+             */
+            var reader = new FileReader();
+            reader.readAsDataURL(blob);
+
+            reader.onloadend = function onLoadEnd(e) {
+                var base64Raw = e.target.result,
+                    commaPosition = base64Raw.indexOf(','),
+                    base64Data = base64Raw.substring(commaPosition + 1),
+                    recording = {
+                        mime: blob.type,
+                        name: filename,
+                        data: base64Data
+                    };
+                self.updateResponse(recording);
+            };
+        },
+
+        /**
+         * Update the PCI response. The recording parameter should follow the QTI 'file' response type
+         * @param {Object} recording
+         * @param {String} recording.mime - mime type of the file
+         * @param {String} recording.name - filename
+         * @param {String} recording.data - base64 encoded file
+         */
+        updateResponse: function updateResponse(recording) {
+            this._recording = recording;
+            if (typeof this.trigger === 'function') {
+                this.trigger('responseChange'); // this has to be camelcase
+            }
+        },
+
+        /**
+         * Update the reset recording button with the number of remaining attempts
+         */
+        updateResetCount: function updateResetCount() {
+            var remaining = this.config.maxRecords - this._recordsAttempts - 1,
+                resetLabel = this.getControlIcon('reset');
+
+            if (this.config.maxRecords > 1) {
+                resetLabel += ' (' + remaining + ')';
+            }
+            if (this.controls.reset) {
+                this.controls.reset.updateLabel(resetLabel);
+            }
+        },
+
+        /**
+         * Add a download link for the recording to the document body (! see inline comment).
+         * This is to be used for testing purposes only.
+         * @param {String} url
+         * @param {String} filename
+         * @param {Number} filesize - in Bytes, size of the recording
+         * @param {Number} durationMs - length of the recording
+         */
+        displayDownloadLink: function displayDownloadLink(url, filename, filesize, durationMs) {
+            var downloadLink;
+
+            if (this.config.displayDownloadLink === true) {
+                downloadLink = document.createElement('a');
+                // fixme: append the link in a better place
+                // container.appendChild(downloadLink); // doesn't work in FF... (nothing happens when link is clicked)
+                document.body.appendChild(downloadLink); // but this works !!!
```

Comment:
I think I did, without more luck. Anyway this is just a testing feature so I think it's acceptable as it is.